scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryNetwork.scala

```diff
@@ -385,6 +381,10 @@ object DiscoveryNetwork {
                 // New response, fold it with the existing to decide if we need more.
                 val next = (acc: Z) => Some(acc -> (count + 1))
                 Task.pure(f(acc, response).bimap(next, next))
+
+              case _ => 
```

Comment:
```suggestion
              case (None, _) => 
```
Makes it easier to understand why.